support-workers/src/typescript/services/payPal.ts

```diff
@@ -0,0 +1,74 @@
+import { z } from 'zod';
+import type { Stage } from '../model/stage';
+import { getConfig } from './config';
+
+const payPalConfigSchema = z.object({
+	user: z.string(),
+	password: z.string(),
+	signature: z.string(),
+	nvpVersion: z.string(),
+	url: z.string(),
+});
+
+type PayPalConfig = z.infer<typeof payPalConfigSchema>;
+
+export const getPayPalConfig = async (stage: Stage): Promise<PayPalConfig> => {
+	return getConfig(stage, 'paypal-config', payPalConfigSchema);
+};
+
+export class PayPalService {
+	defaultNVPParams: Record<string, string>;
+	constructor(private config: PayPalConfig) {
+		this.defaultNVPParams = {
+			USER: config.user,
+			PWD: config.password,
+			SIGNATURE: config.signature,
+			VERSION: config.nvpVersion,
+		};
+	}
+
+	logNVPResponse(response: URLSearchParams) {
+		const msg = (status: string) =>
+			`PayPal: ${status} (NVPResponse: ${response.toString()})`;
+
+		const ack = response.get('ACK');
+		switch (ack) {
+			case 'Success':
+				console.log(msg('Successful PayPal NVP request'));
+				break;
+			case 'SuccessWithWarning':
+				console.log(msg('Warning'));
+				break;
+			case 'Failure':
+				console.log(msg('Error'));
+				break;
+			case 'FailureWithWarning':
+				console.log(msg('Error With Warning'));
+				break;
+		}
+	}
```

Comment:
we can flip this over I think
```suggestion
	logNVPResponse(response: URLSearchParams) {
		const ack = response.get('ACK');
		let status: string = 'unknown';
		switch (ack) {
			case 'Success':
				status = 'Successful PayPal NVP request';
				break;
			case 'SuccessWithWarning':
				status = 'Warning';
				break;
			case 'Failure':
				status = 'Error';
				break;
			case 'FailureWithWarning':
				status = 'Error With Warning';
				break;
		}
		console.log(`PayPal: ${status} (NVPResponse: ${response.toString()})`);
	}
```